File: smartportfolio_cli.py
"""Command-line interface for SmartPortfolio demo."""

from smartportfolio.dashboard import Activity, Portfolio, display_dashboard
from smartportfolio.recommendations import Recommendation, list_recommendations
from smartportfolio.simulation import run_simulation
from smartportfolio.history import show_history
from smartportfolio.settings import Settings, configure_settings


def main() -> None:
    portfolio = Portfolio(
        total_value=10000.0,
        cash_balance=2500.0,
        activities=[
            Activity(description="Initial deposit", amount=10000.0),
        ],
    )
    recommendations = [
        Recommendation(action="buy", ticker="AAPL", reason="Strong earnings"),
        Recommendation(action="sell", ticker="TSLA", reason="High valuation"),
    ]
    history = [9500, 9600, 9700, 9800, 10000]
    settings = Settings()

    while True:
        print("Main Menu")
        print("1. Dashboard")
        print("2. Recommendations")
        print("3. Simulation mode")
        print("4. Portfolio history")
        print("5. Settings")
        print("6. Quit")
        choice = input("Choose an option: ")
        if choice == "1":
            display_dashboard(portfolio)
        elif choice == "2":
            list_recommendations(recommendations)
        elif choice == "3":
            run_simulation(portfolio)
        elif choice == "4":
            show_history(history)
        elif choice == "5":
            settings = configure_settings(settings)
        elif choice == "6":
            break
        else:
            print("Invalid choice.\n")


if __name__ == "__main__":
    main()

File: smartportfolio_cli.js
const readline = require('readline');
const { Activity, Portfolio, displayDashboard } = require('./smartportfolio/dashboard');
const { Recommendation, listRecommendations } = require('./smartportfolio/recommendations');
const { runSimulation } = require('./smartportfolio/simulation');
const { showHistory } = require('./smartportfolio/history');
const { Settings, configureSettings } = require('./smartportfolio/settings');

function createInterface() {
  return readline.createInterface({ input: process.stdin, output: process.stdout });
}

function askQuestion(rl, query) {
  return new Promise(resolve => rl.question(query, answer => resolve(answer)));
}

async function main() {
  const rl = createInterface();
  const ask = q => askQuestion(rl, q);

  let portfolio = new Portfolio(10000.0, 2500.0, [new Activity('Initial deposit', 10000.0)]);
  const recommendations = [
    new Recommendation('buy', 'AAPL', 'Strong earnings'),
    new Recommendation('sell', 'TSLA', 'High valuation')
  ];
  const history = [9500, 9600, 9700, 9800, 10000];
  let settings = new Settings();

  while (true) {
    console.log('Main Menu');
    console.log('1. Dashboard');
    console.log('2. Recommendations');
    console.log('3. Simulation mode');
    console.log('4. Portfolio history');
    console.log('5. Settings');
    console.log('6. Quit');
    const choice = await ask('Choose an option: ');
    if (choice === '1') {
      displayDashboard(portfolio);
    } else if (choice === '2') {
      listRecommendations(recommendations);
    } else if (choice === '3') {
      portfolio = await runSimulation(portfolio, ask);
    } else if (choice === '4') {
      showHistory(history);
    } else if (choice === '5') {
      settings = await configureSettings(settings, ask);
    } else if (choice === '6') {
      break;
    } else {
      console.log('Invalid choice.\n');
    }
  }

  rl.close();
}

if (require.main === module) {
  main();
}
